internal/expr/prepare.go

```diff
@@ -354,23 +294,58 @@ func (pe *ParsedExpr) Prepare(args ...any) (expr *PreparedExpr, err error) {
 				}
 				typeMemberPresent[tm] = true
 			}
+			outputs = append(outputs, typeMembers...)
+		case *bypassPart:
+		default:
+			return nil, fmt.Errorf("internal error: unknown query part type %T", part)
+		}
+	}
+	return &PreparedExpr{inputs: inputs, outputs: outputs, queryParts: pe.queryParts}, nil
+}
 
-			for i, c := range outCols {
+func generateSQL(queryParts []queryPart, sliceLens []int) string {
+	var sql bytes.Buffer
+	var inCount int
+	var outCount int
+	var sliceCount int
+
+	for _, part := range queryParts {
+		switch p := part.(type) {
+		case *inputPart:
+			if p.isSlice {
+				length := defaultSliceLen
+				if sliceLens != nil && sliceLens[sliceCount] > defaultSliceLen {
+					length = sliceLens[sliceCount]
+				}
+				for i := 0; i < length; i++ {
+					sql.WriteString("@sqlair_")
+					sql.WriteString(strconv.Itoa(inCount))
+					if i < length-1 {
+						sql.WriteString(", ")
+					}
+					inCount++
+					sliceCount++
```

Comment:
Should this be outside this inner for loop?

Reply:
Yes it should, I've moved it and added a test for it.